frontend/src/metabase/containers/DataPicker/DataPickerContainer.tsx

```diff
@@ -128,14 +128,16 @@ function DataPicker({
     });
   }, [onChange]);
 
+  const canGoBack = dataTypes.length > 1;
+
   return (
     <DataPickerView
       {...props}
       dataTypes={dataTypes}
       searchQuery={search.query}
       hasDataAccess={hasDataAccess}
       onDataTypeChange={handleDataTypeChange}
-      onBack={handleBack}
+      onBack={canGoBack ? handleReset : undefined}
```

Comment:
The only actual functional change here: we won't offer to go back if there's only one data type available. Felt trivial enough to include here and add a test immediately